x-pack/plugins/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -345,6 +347,67 @@ export class ObservabilityAIAssistantClient {
     });
   };
 
+  get_dataset_info = async (
+    index: string
+  ): Promise<{
+    indices: string[];
+    fields: Array<{ name: string; description: string; type: string }>;
+  }> => {
+    // if empty dataview, get list of all dataviews
+
+    let indices: string[] = [];
+
+    try {
+      const body = await this.dependencies.esClient.indices.resolveIndex({
+        name: index === '' ? '*' : index,
+        expand_wildcards: 'open',
+      });
+      indices = body.indices.map((i) => i.name);
+    } catch (e) {
+      indices = [];
+    }
+
+    if (index === '') {
+      return {
+        indices,
+        fields: [],
+      };
+    }
+
+    if (indices.length === 0) {
+      try {
+        const body = await this.dependencies.esClient.indices.resolveIndex({
+          name: '*',
+          expand_wildcards: 'open',
+        });
+        indices = body.indices.map((i) => i.name);
+      } catch (e) {
+        indices = [];
+      }
+
+      return {
+        indices,
+        fields: [],
+      };
+    }
+
+    const fields = await this.dependencies.dataviews.getFieldsForWildcard({
```

Comment:
should we limit the amount of fields (and indices) that are returned?

Reply:
what would be a reasonable number to limit this to ? i think we also need logic then to get "more" fields as else LLM will assume that the returned fields are the only fields available. Maybe something for a follow up PR ?

Reply:
i added the suggested compression to the field list and limited amount of fields to 500.

here is an issue to track adding chunking in the future:  https://github.com/elastic/kibana/issues/168551